spec/models/user_spec.rb

```diff
@@ -3048,19 +3048,19 @@ def reset_last_seen_cache!(user)
 
       it "tracks old user record correctly" do
         expect do user.update_ip_address!("127.0.0.1") end.to change {
-          UserIpAddressHistory.uncached { UserIpAddressHistory.where(user_id: user.id).count }
```

Comment:
Reverts https://github.com/discourse/discourse/commit/1510e1d1ead8e52eda316978a8df4fda4eead111 where it turns out the root cause of the problem was run.